pkg/lightning/backend/external/writer.go

```diff
@@ -68,6 +69,33 @@ const (
 	DefaultBlockSize = 16 * units.MiB
 )
 
+// GetAdjustedMergeSortOverlapThreshold adjusts the merge sort overlap threshold based on concurrency.
+// The bigger the threshold, the bigger the statistical bias. In CPU:Memory = 1:2 machine, if the concurrency
+// is less than 8, the memory can be used to load data is small, and may get blocked by the memory limiter.
+// So we lower the threshold here if concurrency too low.
+func GetAdjustedMergeSortOverlapThreshold(concurrency int) int64 {
+	intest.Assert(concurrency > 0, "concurrency must be greater than 0, got %d", concurrency)
+	if concurrency < 4 {
+		return 250
+	} else if concurrency < 8 {
+		return 1000
+	} else {
+		return mergeSortOverlapThreshold
+	}
```

Comment:
4000 is the one we used for 16c, so 250 per core, maybe just calculate it by `250 * concurrency`, more smooth